### PR TITLE
test: Make test cases separate functions

### PR DESCRIPTION
**Pull request by maflcko**
This prevents scope-leak of symbols that are supposed to be local to one test case.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Requested by @gwillen in http://www.erisian.com.au/bitcoin-core-dev/log-2019-06-21.html#l-582

**Comment by practicalswift** [comment]
Concept ACK

**Comment by gwillen** [comment]
This seems like a great direction to go! There's probably more we can do in terms of managing state that's shared between test cases, but just explicitly spelling out _which_ state is shared with self.foo, and unsharing other state, is going to remove 90% of the footgun potential right off the top.

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -84,6 +114,7 @@ def run_test(self):
         self.nodes[0].generate(1)
         self.sync_all()
 
+    def test_simple(self):
         ###############
```
Comment (promag) [inline test/functional/rpc_fundrawtransaction.py]: nit, could nuke these "boxes".

**Review by promag — COMMENTED** [review]
ACK 1111451d1, could squash last commit.

**Comment by jonatack** [comment]
Concept ACK, am updating #15687 to this as well after your review comment. Perhaps document where tests are order-dependent, if any. Will review.

**Comment by MarcoFalke** [comment]
Rebased due to conflict in adjacent line

**Pull request merged**